custom_components/coway/sensor.py

```diff
@@ -352,7 +357,94 @@ def available(self) -> bool:
         """Return true if purifier is connected to Coway servers."""
 
         if self.purifier_data.network_status:
-            return True
+            if self.purifier_data.particulate_matter_10.isnumeric():
+                return True
+            else:
+                return False
+        else:
+            return False
+
+class ParticulateMatter25(CoordinatorEntity, SensorEntity):
+    """Representation of PM2.5 measurement."""
+
+    def __init__(self, coordinator, purifier_id):
+        super().__init__(coordinator)
+        self.purifier_id = purifier_id
+
+    @property
+    def purifier_data(self) -> CowayPurifier:
+        """Handle coordinator purifier data."""
+
+        return self.coordinator.data.purifiers[self.purifier_id]
+
+    @property
+    def device_info(self) -> dict[str, Any]:
+        """Return device registry information for this entity."""
+
+        return {
+            "identifiers": {(DOMAIN, self.purifier_data.device_attr['device_id'])},
+            "name": self.purifier_data.device_attr['name'],
+            "manufacturer": "Coway",
+            "model": self.purifier_data.device_attr['model'],
+        }
+
+    @property
+    def unique_id(self) -> str:
+        """Sets unique ID for this entity."""
+
+        return self.purifier_data.device_attr['device_id'] + '_particulate_matter_2_5'
+
+    @property
+    def name(self) -> str:
+        """Return name of the entity."""
+
+        return "Particulate matter 2.5"
+
+    @property
+    def has_entity_name(self) -> bool:
+        """Indicate that entity has name defined."""
+
+        return True
+
+    @property
+    def native_value(self) -> int:
+        """Return current PM2.5 measurement."""
+        pm25 = self.purifier_data.particulate_matter_2_5
+        return pm25 if pm25.isnumeric() else 0
```

Comment:
If the IconS returns PM25 values, then please confirm if a non-empty string is ever returned - I suspect this isn't the case. In addition, the check you have included in the `available` property would set the sensor to unavailable making the isnumeric check here redundant (a state of 0 will never be recorded).

Reply:
right, I haven't seen a non-empty string